### PR TITLE
HiDPI cursors: Support Qt < 5

### PR DESCRIPTION
**Pull request by Vanuan**
Forum thread: https://forum.freecadweb.org/viewtopic.php?p=433621#p433621

### CODE REVIEW

File: src/Mod/Draft/draftguitools/gui_snapper.py [hunk 1]
```diff
@@ -1122,7 +1122,8 @@ def device_pixel_ratio(self):
         mw = Gui.getMainWindow()
         for w in mw.findChild(QtGui.QMdiArea).findChildren(QtGui.QWidget):
             if w.metaObject().className() == "SIM::Coin3D::Quarter::QuarterWidget":
-                device_pixel_ratio = w.devicePixelRatio()
+                if int(QtCore.qVersion()) > 4:
+                  device_pixel_ratio = w.devicePixelRatio()
```
Comment (vocx-fc) [inline src/Mod/Draft/draftguitools/gui_snapper.py:1126]: Please use 4-space indentation.

File: src/Mod/Draft/draftguitools/gui_snapper.py [hunk 2]
```diff
@@ -1146,7 +1147,8 @@ def get_cursor_with_tail(self, base_icon_name, tail_icon_name=None):
         qp.end()
         cur_hot_x = 0.25 * full_icon_size * device_pixel_ratio
         cur_hot_y = 0.25 * full_icon_size * device_pixel_ratio
-        new_icon.setDevicePixelRatio(device_pixel_ratio)
+        if int(QtCore.qVersion()) > 4:
+          new_icon.setDevicePixelRatio(device_pixel_ratio)
```
Comment (vocx-fc) [inline src/Mod/Draft/draftguitools/gui_snapper.py:1151]: Please use 4-space indentation.

File: src/Mod/Draft/draftguitools/gui_snapper.py (hunk 1 above)
Comment (Vanuan) [reply src/Mod/Draft/draftguitools/gui_snapper.py]: done

File: src/Mod/Draft/draftguitools/gui_snapper.py (hunk 2 above)
Comment (Vanuan) [reply src/Mod/Draft/draftguitools/gui_snapper.py]: done

**Review by vocx-fc — APPROVED** [review]

**Comment by vocx-fc** [comment]
@Vanuan I just noticed that this causes a problem. `QtCore.qVersion()` returns a string with semantic versioning like `5.9.5`.

Therefore, the first character needs to be parsed to identify whether it is Qt4 or Qt5.
```py
if int(QtCore.qVersion()[0]) > 4:
```

Or we can do string comparison directly. This works in Python because it compares the ASCII values, and '4' is less than '5.9.5'.
```py
if QtCore.qVersion() > '4':
```

**Comment by Vanuan** [comment]
> Or we can do string comparison directly. This works in Python because it compares the ASCII values, and '4' is less than '5.9.5'.

Not future proof:

```
>>> "10.0.0" > "4"
False

```

**Comment by Vanuan** [comment]
Maybe something like #3909 ?

**Comment by vocx-fc** [comment]
LOL, yeah. Qt will probably take a while to reach 10.0. Who knows what will happen in between. `qVersion` may not even exist. But sure your new version works as well.

**Pull request merged**